packages/angular/src/schematics/init/init.ts

```diff
@@ -0,0 +1,369 @@
+      `Project ${config.name} is missing serve and/or build targets`
+    );
+  }
+
+  const build = config.targets['build'];
+  const serve = config.targets['serve'];
+
+  return { build, serve, projectName, config, sourceRoot, root };
+}
+
+export default convertNxGenerator(ngAddGenerator);
+
+export function throwIfNullish<T>(
+  value: T | undefined,
+  message = 'Value is nullish'
+): T {
+  if (value === undefined || value === null) {
+    throw new Error(message);
+  }
+  return value;
+}
+
+export function assertNotNullish<T>(value: T | undefined): asserts value is T {
+  throwIfNullish(value);
+}
+
+/**
+ * Copy from https://github.com/nrwl/nx/blob/master/packages/create-nx-workspace/src/utils/package-manager.ts#L21
+ * Cannot use original from nx because we need to access the try and not the local file system for testing.
+ */
+
+export function detectPackageManager(
+  tree: Tree
+): 'bun' | 'yarn' | 'pnpm' | 'npm' {
+  if (tree.exists('bun.lockb') || tree.exists('bun.lock')) {
+    return 'bun';
+  }
+  if (tree.exists('yarn.lock')) {
+    return 'yarn';
+  }
+  if (tree.exists('pnpm-lock.yaml')) {
+    return 'pnpm';
+  }
+  return 'npm';
+}
+
+/**
+ * Ensures Playwright is installed with a compatible version.
+ * Checks if it's already installed and warns if the version is incompatible.
+ * If not installed, adds it to package.json with the required version and
+ * runs the install task of the existing package manager.
+ */
+export function ensurePlaywrightIsInstalled(tree: Tree): void {
+  const requiredRange = getRequiredPlaywrightRange();
+  const installedVersion = getInstalledPlaywrightVersion(tree);
+
+  if (installedVersion) {
+    if (
+      !isVersionCompatible(
+        installedVersion,
+        requiredRange.upper,
+        requiredRange.lower
+      )
+    ) {
+      logger.warn(
+        `Installed Playwright version (${installedVersion}) may not be compatible with Testronaut. ` +
+          `Recommended version: ${requiredRange.upper}. Consider changing your Playwright version to avoid issues.`
+      );
+    }
+  } else {
+    addDependenciesToPackageJson(
+      tree,
+      {},
+      { '@playwright/test': requiredRange.upper }
+    );
+    installPackagesTask(tree);
+  }
+}
+
+/**
+ * Provides the required Playwright version range for Testronaut.
+ */
+function getRequiredPlaywrightRange() {
+  return PLAYWRIGHT_VERSION_RANGE;
+}
+
+export function parseMaxSupportedVersion(versionRange: string) {
+  // Strict format: >=x.y.z <=a.b.c (with at least one space)
+  // Both versions must have major.minor.patch format
+  const regex = /^>=([\d.]+)\s+<=([\d.]+)$/;
+  const match = versionRange.match(regex);
+
+  if (!match) {
+    throw new Error(
+      `Invalid version range format: ${versionRange}. Must use format ">=x.y.z <=a.b.c" with major.minor.patch versions.`
+    );
+  }
+
+  // Extract both bounds
+  const lower = match[1];
+  const upper = match[2];
+
+  return { lower, upper };
+}
+
+/**
```

Comment:
I don't know why `@testronaut/core` is a peer dependency. I think that it could have been a direct dependency.
Concerning `@playwrihgt/test`, the package is actually used by both `@testronaut/angular` and `@testronaut/core` so we need it there.